docs/How_To/Tutorials/document_reader.rst

```diff
@@ -0,0 +1,253 @@
+
+
+    class DocumentSearch(BaseAction):
+        def __init__(self, filename, embedding_function, k=4):
+            # file name of the pdf
+            self.filename = filename
+            # the embedding function to use
+            self.embedding_function = embedding_function
+            # number of results to return in search
+            self.k = k
+
+            # load the pdf and create the vector store
+            self.chroma = Chroma(embedding_function = embedding_function)
+            documents = PDFMinerLoader(self.filename).load()
+            documents = SentenceTransformersTokenTextSplitter(chunk_overlap=0).split_documents(documents)
+
+            logger.info(f"Adding {len(documents)} documents to the vector store")
+            self.chroma.add_documents(documents)
+            logger.info("Finished adding documents to the vector store")
+
+        def execute(self, query):
+            """
+            Execute the action by searching the document store for the query
+
+            Args:
+                query (str): The query to search for
+
+            Returns:
+                str: The search results combined into a single string
+            """
+
+            results = self.chroma.search(query, search_type="mmr", k=self.k)
+            return "\n\n".join([result.page_content for result in results])
+
+        @property
+        def name(self) -> str:
+            """
+            The name of the action, used to describe the action to the agent.
+            """
+            return "DocumentSearch"
+        
+        @property
+        def args(self) -> dict:
+            """
+            The arguments that the action takes, used to describe the action to the agent.
+            """
+            return {
+                "query": "string"
+            }
+
+The action is a crucial part of Sherpa enabling the agent to interact with any other systems. In this case, the action is used for searching the vector database containing PDF content for the query.
+
+This action will be passed to the QAAgent to enable the agent to search the vector database for the query.
+
+There are three main parts in this action class:
+
+1. `__init__`: This method initializes the action by loading the PDF file, creating the vector database and adding the documents to the vector database.
+
+2. `execute`: This method executes the action by searching the vector database for the query and returning the search results.
+
+3. `name` and `args`: These properties are used to describe the action to the agent.
+
+
+Find a PDF file
+****************
+
+Next, you can find a PDF file to use with the PDF reader. For this tutorial, we will use a paper PDF as an example. You can download the PDF file from the following link: https://arxiv.org/pdf/2401.07930.pdf. You can also use any other PDF file you have. Save the PDF file in the same folder as the `actions.py` file. For simplicity, we will use the filename `paper.pdf` in this tutorial.
```

Comment:
this is nice, but as we discussed I'm hoping for the simplest non-trivial example. can we add another action like google search? then we would have two actions which makes it nontrivial

Reply:
I added a new section to the tutorial about Google search and the verification steps